d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerTest.java

```diff
@@ -155,6 +161,87 @@ public void doOneTimeTearDown() throws IOException
     }
   }
 
+  @DataProvider
+  public Object[][] provideKeys()
+  {
+    return new Object[][] {
+        // numHttp, numHttps, expectedNumHttp, expectedNumHttps, partitionIdForAdd, partitionIdForCheck
+        {0, 3, 0, 3, 0, 0},
+        {3, 0, 3, 0, 0, 0},
+        {1, 1, 1, 1, 0, 0},
+        {0, 0, 0, 0, 0, 0},
+        // alter the partitions to check
+        {0, 3, 0, 0, 0, 1},
+        {3, 0, 0, 0, 0, 1},
+        {1, 1, 0, 0, 0, 2},
+        {0, 0, 0, 0, 0, 1},
+        // alter the partitions to add and check to match
+        {0, 3, 0, 3, 1, 1},
+        {3, 0, 3, 0, 1, 1},
+        {1, 1, 1, 1, 2, 2},
+        {0, 0, 0, 0, 1, 1}
+    };
+  }
+
+  @Test(dataProvider = "provideKeys")
+  public void testClusterInfoProvider(int numHttp, int numHttps, int expectedNumHttp, int expectedNumHttps,
+      int partitionIdForAdd, int partitionIdForCheck)
+      throws InterruptedException, ExecutionException, ServiceUnavailableException
+  {
+    Map<String, LoadBalancerStrategyFactory<? extends LoadBalancerStrategy>> loadBalancerStrategyFactories =
+        new HashMap<String, LoadBalancerStrategyFactory<? extends LoadBalancerStrategy>>();
+    Map<String, TransportClientFactory> clientFactories =
+        new HashMap<String, TransportClientFactory>();
+
+    MockStore<ServiceProperties> serviceRegistry = new MockStore<ServiceProperties>();
```

Comment:
-> new MockStore<>()

Here and everywhere else

Reply:
I'm only gonna modify this function. Doing it for all others is a cleanup task that shouldn't be done with this RB, it'd obfuscate what I really added here.